app/vmctl/opentsdb/opentsdb.go

```diff
@@ -0,0 +1,279 @@
+// e.g. /api/search/lookup?m=system.load5&limit=1000000
+func (c Client) FindSeries(metric string) ([]Meta, error) {
+	q := fmt.Sprintf("%s/api/search/lookup?m=%s&limit=%d", c.Addr, metric, c.Limit)
+	resp, err := http.Get(q)
+	if err != nil {
+		return nil, fmt.Errorf("Could not properly make request to %s: %s", c.Addr, err)
+	}
+	defer func() { _ = resp.Body.Close() }()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("Could not retrieve series data from %s: %s", c.Addr, err)
+	}
+	var results MetaResults
+	err = json.Unmarshal(body, &results)
+	if err != nil {
+		return nil, fmt.Errorf("Invalid series data from %s: %s", c.Addr, err)
+	}
+	return results.Results, nil
+}
+
+// GetData actually retrieves data for a series at a specified time range
+func (c Client) GetData(series Meta, rt Retention, start int64, end int64) (Metric, error) {
+	expr := Expression{}
+	expr.Outputs = append(expr.Outputs, outputObj{ID: "a", Alias: "query"})
+	expr.Metrics = append(expr.Metrics, metricObj{ID: "a", Metric: series.Metric,
+		Filter: "f1", FillPolicy: fillObj{Policy: "nan"}})
+	expr.Time = timeObj{Start: start, End: end, Aggregator: rt.FirstOrder,
+		Downsampler: dSObj{Interval: rt.AggTime,
+			Aggregator: rt.SecondOrder,
+			FillPolicy: fillObj{Policy: "nan"}}}
+	var TagList []tagObj
+	for k, v := range series.Tags {
+		TagList = append(TagList, tagObj{Type: "literal_or", Tagk: k,
+			Filter: v, GroupBy: true})
+	}
+	expr.Filters = append(expr.Filters, filterObj{ID: "f1", Tags: TagList})
+	// "expressions" is required in the query object or we get a 5xx, so force it to exist
+	expr.Expressions = make([]int, 0)
+	inputData, err := json.Marshal(expr)
+	if err != nil {
+		return Metric{}, fmt.Errorf("Failed to marshal query results %v", err)
+	}
+	// log.Println("Query: ", string(inputData))
+	q := fmt.Sprintf("%s/api/query/exp", c.Addr)
+	resp, err := http.Post(q, "application/json", bytes.NewBuffer(inputData))
+	if err != nil {
+		return Metric{}, fmt.Errorf("Could not properly make request to %s: %s", c.Addr, err)
+	}
+	defer func() { _ = resp.Body.Close() }()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return Metric{}, fmt.Errorf("Could not retrieve series data from %s: %s", c.Addr, err)
+	}
+	// log.Println(fmt.Sprintf("Initial Output: %v", string(body)))
+	var output ExpressionOutput
+	err = json.Unmarshal(body, &output)
+	if err != nil {
+		// log.Println("Incoming data: ", string(body))
+		return Metric{}, fmt.Errorf("Invalid series data from %s: %s", c.Addr, err)
+	}
+	if len(output.Outputs) < 1 {
+		// log.Println("Incoming data: ", string(body))
+		return Metric{}, nil
+	}
+	// log.Println("De-serialized: ", output)
+	data := Metric{}
+	data.Metric = series.Metric
+	data.Tags = series.Tags
+	for _, tsobj := range output.Outputs[0].Dps {
+		data.Timestamps = append(data.Timestamps, int64(tsobj[0]))
+		data.Values = append(data.Values, tsobj[1])
+	}
+	data, err = modifyData(data, c.Normalize)
```

Comment:
A lot of conversions happens here. I think, it is worth to describe data model mapping in the README

Reply:
Basically, we're just converting from OpenTSDB's format to VictoriaMetrics format.